fdbserver/ptxn/TLogServer.actor.cpp

```diff
@@ -1123,18 +1122,18 @@ ACTOR Future<Void> tLogCommit(Reference<TLogGroupData> self,
 	return Void();
 }
 
-Optional<std::pair<Version, std::pair<StringRef, Arena>>> LogGenerationData::getSerializedTLogData(
+Optional<std::pair<Version, Standalone<StringRef>>> LogGenerationData::getSerializedTLogData(
     const Version& version,
     const StorageTeamID& storageTeamID) {
 
 	auto pStorageTeamData = getStorageTeamData(storageTeamID);
 	// by lower_bound, if we pass in 10, we might get 12, and return 12
 	auto iter = pStorageTeamData->versionMessages.lower_bound(version);
 	if (iter == pStorageTeamData->versionMessages.end()) {
-		return Optional<std::pair<Version, std::pair<StringRef, Arena>>>();
+		return Optional<std::pair<Version, Standalone<StringRef>>>();
 	}
-
-	return std::make_pair(iter->first, iter->second);
+	Standalone<StringRef> data = Standalone(iter->second.first);
```

Comment:
I think this does a copy. A better way is:`Standalone<StringRef> data(iter->second.first, iter->second.second)`, i.e., using `Standalone(const T& t, const Arena& arena)` constructor.

BTW, `pStorageTeamData->versionMessages` uses type `std::map<Version, std::pair<StringRef, Arena>>`, and I think you can refactor its type as well.

Reply:
sounds good, yeah I was wondering if that is a good idea, will do